// nav-frontend-internheader/src/brukermeny.tsx
import * as React from 'react';
import { Flatknapp } from 'nav-frontend-knapper';
import { Normaltekst } from 'nav-frontend-typografi';
import * as PT from 'prop-types';

interface IProps {
    /**
     * Navn på saksbehandler.
     */
    navn: string;

    /**
     * Enheten saksbehandler tilhører.
     */
    enhet?: string;

    /**
     * Identen til saksbehandler.
     */
    ident?: string;

    /**
     * Tittelen til saksbehandler.
     * F.eks. veilder.
     */
    tittel?: string;

    children?: React.ReactChildren;
}

class Brukermeny extends React.Component<IProps> {
    render() {
        const { enhet, ident, navn, children, tittel } = this.props;

        return (
            <div className={'nav-frontend-internheader__brukermeny'}>
                <div
                    className={
                        'nav-frontend-internheader__brukermeny--saksbehandler'
                    }
                >
                    <Normaltekst
                        className={
                            'nav-frontend-internheader__brukermeny--saksbehandler-tekst'
                        }
                    >
                        {`${navn}${ident ? `  ${ident}` : ''}`}
                    </Normaltekst>
                    <Normaltekst
                        className={
                            'nav-frontend-internheader__brukermeny--saksbehandler-tekst'
                        }
                    >
                        {`${enhet ? enhet : ''}${
                            tittel ? (enhet ? `, ${tittel}` : tittel) : ''
                        }`}
                    </Normaltekst>
                </div>

                {children}
            </div>
        );
    }
}

(Brukermeny as React.ComponentClass).propTypes = {
    /**
     * Egendefinert klassenavn.
     */
    className: PT.string,

    navn: PT.string,

    /**
     * Enheten saksbehandler tilhører.
     */
    enhet: PT.string,

    /**
     * Identen til saksbehandler.
     */
    ident: PT.string,

    /**
     * Tittelen til saksbehandler.
     * F.eks. veilder.
     */
    tittel: PT.string,

    onLoggUtClick: PT.func
};

export default Brukermeny;
