// nav-frontend-internheader/src/internheader.tsx
import * as React from 'react';
import * as PT from 'prop-types';
import * as classnames from 'classnames';
import { Systemtittel, Normaltekst } from 'nav-frontend-typografi';
import { Flatknapp } from 'nav-frontend-knapper';
import 'nav-frontend-internheader-style';

export interface NavFrontendInternheaderProps {
    /**
     * Egendefinert klassenavn.
     */
    className?: string;

    onProduktTittelClick?: () => void;

    /**
     * Tittel på produkt.
     * F.eks. NAV Kontantstøtte
     */
    produktTittel: string;

    /**
     * Visningsnavn tilknyttet saksbehandler
     */
    saksbehandlerNavn: string;

    /**
     * NAV ident tilknyttet saksbehandler.
     */
    saksbehandlerIdent?: string;

    /**
     * Enhet tilknyttet saksbehandler.
     * F.eks. NAV Sagene
     */
    saksbehandlerEnhet: string;

    /**
     * Tittel tilknyttet saksbehandler.
     * F.eks. veilder
     */
    saksbehandlerTittel: string;

    onLoggUtClick?: () => void;
}

class NavFrontendInternheader extends React.Component<
    NavFrontendInternheaderProps
> {
    render() {
        const {
            className,
            onLoggUtClick,
            onProduktTittelClick,
            produktTittel,
            saksbehandlerEnhet,
            saksbehandlerIdent,
            saksbehandlerNavn,
            saksbehandlerTittel
        } = this.props;
        return (
            <div className={classnames('nav-frontend-internheader', className)}>
                <div className={'nav-frontend-internheader__produkttittel'}>
                    <HeaderKnapp onClick={onProduktTittelClick}>
                        <Systemtittel>{produktTittel}</Systemtittel>
                    </HeaderKnapp>

                    <Divider margin={'0 1rem 0 2rem'} />
                </div>

                <div className={'nav-frontend-internheader__brukermeny'}>
                    <Divider margin={'0 2rem 0 0'} />
                    <div
                        className={
                            'nav-frontend-internheader__brukermeny--saksbehandler'
                        }
                    >
                        <Normaltekst
                            className={
                                'nav-frontend-internheader__brukermeny--saksbehandler-tekst'
                            }
                        >
                            {`${saksbehandlerNavn}${saksbehandlerIdent &&
                                `  ${saksbehandlerIdent}`}`}
                        </Normaltekst>
                        <Normaltekst
                            className={
                                'nav-frontend-internheader__brukermeny--saksbehandler-tekst'
                            }
                        >
                            {`${saksbehandlerEnhet}${saksbehandlerTittel &&
                                `, ${saksbehandlerTittel}`}`}
                        </Normaltekst>
                    </div>

                    <HeaderKnapp onClick={onLoggUtClick}>Logg ut</HeaderKnapp>
                </div>
            </div>
        );
    }
}

const HeaderKnapp: React.StatelessComponent<{
    onClick?: () => void;
    children: React.ReactNode;
}> = ({ children, onClick }) => {
    return (
        <div
            className={'nav-frontend-internheader__knapp'}
            onClick={onClick}
            role={'button'}
        >
            {children}
        </div>
    );
};

const Divider: React.StatelessComponent<{ margin?: string }> = ({ margin }) => {
    return (
        <div
            className={'nav-frontend-internheader__divider'}
            style={{ margin }}
        />
    );
};

(NavFrontendInternheader as React.ComponentClass).propTypes = {
    /**
     * Egendefinert klassenavn.
     */
    className: PT.string,

    produktTittel: PT.string
};

export default NavFrontendInternheader;
